contracts/TokenDistributor.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import {StakingRewardsV2} from "./StakingRewardsV2.sol";
+import {RewardEscrowV2} from "./RewardEscrowV2.sol";
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {ITokenDistributor} from "./interfaces/ITokenDistributor.sol";
+
+contract TokenDistributor is ITokenDistributor {
+    /// @notice represents the status of if a person already
+    /// claimed their epoch
+    mapping(address => mapping(uint => bool)) internal claimedEpochs;
+
+    /// @notice kwenta interface
+    IKwenta private kwenta;
+
+    /// @notice rewards staking contract
+    StakingRewardsV2 private stakingRewardsV2;
+
+    /// @notice escrow contract which holds (and may stake) reward tokens
+    RewardEscrowV2 private rewardEscrowV2;
+
+    /// @notice last recorded balance of KWENTA in contract
+    uint internal lastTokenBalance;
+
+    /// @notice last checkpoint time
+    uint internal lastCheckpoint;
+
+    /// @notice starting week of deployment
+    uint internal startTime;
+
+    /// @inheritdoc ITokenDistributor
+    mapping (uint => uint) public tokensPerEpoch;
```

Comment:
Group your mappings together, also could make sense to group your variables by visibility. It took me a while to catch that this variable is the only one that's public.